### ISSUE TITLE
Object mode

### ISSUE THREAD

**Issue by tmcw** [placemark/check-geojson#17]
There should be a mode that works on JSON.parse-generated objects without the AST overhead.

**Comment by stevage** [comment]
In my current use case, I'm generating GeoJSON objects natively in my code. I'm using this library to check for bugs in that logic, essentially. So currently I'm calling JSON.stringify first.

**Comment by stevage** [comment]
Hmm, now that I've looked at the code, this sure isn't trivial to implement. I'm not sure it's viable? The whole code base is traversing an AST (abstract syntax tree, for passers-by), rather than the object itself. For instance:

```
const typeMember = node.members.find((member) => {
    return member.name.value === 'type';
  });
```


To make this work on a native object would seem to require one of these approaches:

1. Simply call JSON.stringify() first, and add warnings to the README. (With big performance hit.)
2. Build a fake AST from the object first, then proceed as normal. (With performance hit.)
3. Modify every function to work with either an AST or native object. (With readability/maintainability cost, plus implementation effort.)
4. Duplicate the codebase to provide parallel versions of every function. (With maintainability cost, plus implementation effort.)

Alternatively, option 5, is just to make clearer that this is a library dedicated to parsing and validating human-written GeoJSON, and maybe throw in a nice error when given an object rather than a string.

I suspect option 5 is better than 1: so the dev sees the performance warning, then can make a conscious choice.

Otherwise, perhaps option 2 is not sooo bad?

**Comment by tmcw** [comment]
Yeah… I think the right solution in the short term is clarify in the readme that this expects a string, and do a runtime check that it's receiving a string. My tendency has been to avoid runtime checks because documentation + types should describe expected inputs & doing runtime type checking is redundant. But probably not everyone is using TypeScript or has good types.

I perused a few functions and I think _maybe_ the way to do it would be to add some helper functions (sort of like turf/meta) that let this work with either the AST or raw types, but yeah, it's a close race between the options and none of them are that quick.
